chalice/app.py

```diff
@@ -264,17 +273,40 @@ def __init__(self, body, headers=None, status_code=200):
         if headers is None:
             headers = {}
         self.headers = headers
+        self.case_insensitive_headers = CaseInsensitiveMapping(headers)
         self.status_code = status_code
+        self._is_base64_encoded = False
+
+    def _base64encode(self, data):
+        if not isinstance(data, bytes):
+            raise ValueError('Expected bytes type for body with binary '
+                             'Content-Type. Got %s type body instead.'
+                             % type(data))
+        data = base64.b64encode(data)
+        return data.decode('ascii')
+
+    def encode_as_binary(self):
```

Comment:
So I read the comment, looked at `test_invalid_binary_response_body_throws_value_error`, and I still don't understand why this is needed.  I suspect customers will also be confused what this public method does and why it's here.